File: clinic-admin/src/api/medicalRecord.ts
import { get, post } from '@/utils/request'

export interface MedicalRecord {
  id?: number
  recordId?: number
  recordNo: string
  patientId: number
  patientName: string
  patientPhone: string
  registrationId: number
  doctorId: number
  doctorName: string
  deptId: number
  deptName: string
  visitType: number
  visitTypeName: string
  visitDate: string
  chiefComplaint: string
  diagnosis: string
  treatment: string
  medicalAdvice: string
  attachUrls: string
  createTime: string
  updateTime: string
}

export interface MedicalRecordQuery {
  clinicId: number
  patientId?: number
  doctorId?: number
  startDate?: string
  endDate?: string
  keyword?: string
  pageNum?: number
  pageSize?: number
}

export const getMedicalRecordList = (params: MedicalRecordQuery) =>
  get<{ total: number; list: MedicalRecord[] }>('/admin/medicalRecord/list', params)

export const getMedicalRecordDetail = (recordId: number) =>
  get<MedicalRecord>(`/admin/medicalRecord/${recordId}`)

export const createMedicalRecord = (data: Partial<MedicalRecord>) =>
  post('/admin/medicalRecord/create', data)

export const updateMedicalRecord = (data: Partial<MedicalRecord>) =>
  post('/admin/medicalRecord/update', data)

export const deleteMedicalRecord = (recordId: number) =>
  post('/admin/medicalRecord/delete', { recordId })

File: clinic-api/src/main/java/com/clinic/entity/MedicalRecord.java
package com.clinic.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("medical_record")
public class MedicalRecord implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long clinicId;
    private String recordNo;
    private Long patientId;
    private String patientName;
    private String patientPhone;
    private Long registrationId;
    private Long doctorId;
    private String doctorName;
    private Long deptId;
    private String deptName;
    private Integer visitType;
    private String chiefComplaint;
    private String diagnosis;
    private String treatment;
    private String medicalAdvice;
    //private String attachUrls;
    //private LocalDateTime createTime;
    //private LocalDateTime updateTime;
    private LocalDate visitDate;

    @TableField(exist = false)
    private String visitTypeName;

    @TableField(exist = false)
    private List<ToothChartRecord> toothChartRecords;

    @TableField(exist = false)
    private String toothChartSummary;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}